src/storage/object.ts

```diff
@@ -447,4 +448,39 @@ export class ObjectStorage {
       })
     )
   }
+
+  /**
+   * Generates a signed url for uploading an object
+   * @param objectName
+   * @param url
+   * @param expiresIn
+   * @param metadata
+   * @returns
+   */
+  async signUploadObjectUrl(
+    objectName: string,
+    url: string,
+    expiresIn: number,
+    metadata?: Record<string, string>
+  ) {
+    await this.db.findBucketById(this.bucketId) // Require bucket to be present
+
+    let found
+    try {
+      found = await this.findObject(objectName)
+    } catch (e) {
+      // Don't do anything, since it will throw an error if the object doesn't exist (but that is what we want.)
```

Comment:
This is not going to work reliably,
If we have certain RLS policies defined this might return 404 because we don't have access to the file not because it doesn't exist

We might want to run this query `asSuperUser` to make sure the file is not present at this location.

However, when doing this as superUser we don't have any protection on who can create these upload URL.

As a second step, after the file existence, we need to check if the user has the CREATE RLS permission in order to generate the upload url